homeassistant/components/recorder/__init__.py

```diff
@@ -494,6 +555,14 @@ def run(self):
 
         # Start periodic purge
         if self.auto_purge:
+            # Compile daily statistics every night at 12:12am
+            track_time_change(
```

Comment:
Would be slightly more efficient to move these into `def _async_setup_interval_tasks`, change them to `async_track_time_change` and call `self._async_setup_interval_tasks` with `run_callback_threadsafe` so we don't have to schedule 3 callbacks in the event loop.

Reply:
Fixed in 6b01c78b5dd7896eb08645bf1e0c4217278f3123